aspnetcore/mvc/models/validation/samples/3.x/ValidationSample/Pages/Movies/Create.cshtml.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using ValidationSample.Data;
+using ValidationSample.Models;
+
+namespace ValidationSample.Pages.Movies
+{
+    public class CreateModel : PageModel
+    {
+        private readonly MovieContext _context;
+
+        public CreateModel(MovieContext context)
+        {
+            _context = context;
+        }
+
+        [BindProperty]
+        public Movie Movie { get; set; }
+
+        #region snippet_OnPostAsync
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            _context.Movies.Add(Movie);
+            await _context.SaveChangesAsync();
+
+            return RedirectToPage("./Index");
+        }
+        #endregion
+
+        public async Task<IActionResult> OnPostTryValidateAsync()
+        {
+            var modifiedReleaseDate = DateTime.Now.Date;
+
+            #region snippet_TryValidate
+            Movie.ReleaseDate = modifiedReleaseDate;
+
+            if (!TryValidateModel(Movie, nameof(Movie)))
```

Comment:
The 2.x sample called `TryValidateModel` without checking the return value and then checking `ModelState.IsValid` straight after, but this approach achieves both in a simpler manner.

---

For MVC, this would be more like:

```csharp
if (!TryValidateModel(movie)) // model-bound movie action parameter
```
There's a question that keeps coming up: What's the best way to show both approaches in the docs page?